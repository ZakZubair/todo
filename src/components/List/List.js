import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import Item from '../Item';
import { getFormattedTime } from '../../helpers';

class List extends PureComponent {
  render() {
    const {
      listItems,
      onToggle,
      onEdit,
      onRemove,
    } = this.props;

    const todoList = listItems && listItems.length > 0 && listItems.map(
      ({
        id,
        name,
        description,
        checked,
        dateCreated,
        dateUpdated,
      }) => (
        <Item
          id={id}
          name={name}
          description={description}
          date={getFormattedTime(dateCreated)}
          dateUpdated={dateUpdated ? getFormattedTime(dateUpdated) : ''}
          checked={checked}
          onToggle={onToggle}
          onRemove={onRemove}
          onEdit={onEdit}
          key={id}
        />
      ),
    );

    return (
      <div className="list-container">
        {todoList && todoList}
      </div>
    );
  }
}

List.defaultProps = {
  listItems: [],
};

List.propTypes = {
  listItems: PropTypes.arrayOf(PropTypes.any),
  onToggle: PropTypes.func.isRequired,
  onRemove: PropTypes.func.isRequired,
  onEdit: PropTypes.func.isRequired,
};
export default List;
