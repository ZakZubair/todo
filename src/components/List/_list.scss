.list-container {
  background-color: $white;
  box-shadow: 0 rem(8) rem(16) 0 $mischka-35;
  text-shadow: 0 rem(8) rem(16) $mischka-35;
}
