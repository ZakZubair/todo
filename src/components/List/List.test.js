import React from 'react';
// import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import List from './List';

describe('test List component', () => {
  it('render correctly list component', () => {
    const mockCallBack = jest.fn();

    const props = {
      listItems: [
        {
          id: 'abc123',
          name: 'Test name',
          description: 'Test description',
          date: 'Date time ',
          checked: false,
        },
        {
          id: 'abc456',
          name: 'Test name 2',
          description: 'Test description 2',
          date: 'Date time ',
          checked: false,
          dateUpdated: new Date(),
        },
      ],
      onToggle: mockCallBack,
      onRemove: mockCallBack,
      onEdit: mockCallBack,
    };

    const ListComponent = renderer.create(<List {...props} />).toJSON();
    expect(ListComponent).toMatchSnapshot();
  });
});
