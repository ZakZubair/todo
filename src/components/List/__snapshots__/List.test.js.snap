// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test List component render correctly list component 1`] = `
<div
  className="list-container"
>
  <div
    className="item"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    tabIndex="abc123"
  >
    <div
      className="item__check-mark"
    />
    <div
      className="item__text "
    >
      <p
        className="item__text__name"
      >
        <span>
          Test name
        </span>
      </p>
      <p
        className="item__text__description"
      >
        Test description
      </p>
      <p
        className="item__text__date"
      >
        Created at 13/04/2020 4:28 pm
      </p>
    </div>
    <button
      className="button__edit"
      onClick={[Function]}
      type="button"
    >
      Edit
    </button>
    <button
      className="button__remove"
      onClick={[Function]}
      type="button"
    >
      ×
    </button>
  </div>
  <div
    className="item"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    tabIndex="abc456"
  >
    <div
      className="item__check-mark"
    />
    <div
      className="item__text "
    >
      <p
        className="item__text__name"
      >
        <span>
          Test name 2
        </span>
      </p>
      <p
        className="item__text__description"
      >
        Test description 2
      </p>
      <p
        className="item__text__date"
      >
        Updated at 13/04/2020 4:28 pm
      </p>
    </div>
    <button
      className="button__edit"
      onClick={[Function]}
      type="button"
    >
      Edit
    </button>
    <button
      className="button__remove"
      onClick={[Function]}
      type="button"
    >
      ×
    </button>
  </div>
</div>
`;
